api/debug/flags.go

```diff
@@ -136,37 +186,71 @@ func CreateLogDir(logDir string) {
 // Setup initializes profiling and logging based on the CLI flags.
 // It should be called as early as possible in the program.
 func Setup(ctx *cli.Context) error {
+	var ostream log.Handler
+	output := io.Writer(os.Stderr)
+	if ctx.GlobalBool(logjsonFlag.Name) {
+		ostream = log.StreamHandler(output, log.JsonFormat())
+	} else {
+		usecolor := (isatty.IsTerminal(os.Stderr.Fd()) && os.Getenv("TERM") != "dumb")
+		if usecolor {
+			output = colorable.NewColorableStderr()
+		}
+		ostream = log.StreamHandler(output, log.TerminalFormat(usecolor))
+	}
+	glogger.SetHandler(ostream)
+
 	// logging
-	log.PrintOrigins(ctx.GlobalBool(debugFlag.Name))
+	log.PrintOrigins(ctx.GlobalBool(getFlagName(ctx, debugFlag, legacyDebugFlag, "(v1.12.0)")))
 	log.ChangeGlobalLogLevel(glogger, log.Lvl(ctx.GlobalInt(verbosityFlag.Name)))
 	glogger.Vmodule(ctx.GlobalString(vmoduleFlag.Name))
-	glogger.BacktraceAt(ctx.GlobalString(backtraceAtFlag.Name))
+
+	if location := ctx.GlobalString(getFlagName(ctx, backtraceAtFlag, legacyBacktraceAtFlag, "(v1.12.0)")); location != "" {
+		glogger.BacktraceAt(location)
+	}
```

Comment:
@hyochan-brown `glogger.BacktraceAt` is not executed when `location` is "". Is this intended? if this is not intended, could you check other flags?

Reply:
You're right.
I thought flag without value should not be executed.
So setting the flag to `--log.backtrace (no value)` does not executed.
Did I make a mistake..?

Reply:
@hyochan-brown If it is intended, it's okay :)